apps/ui/src/views/Editor.vue

```diff
@@ -339,7 +341,11 @@ export default defineComponent({
         <UiLinkPreview :key="proposalKey || ''" :url="proposal.discussion" />
       </div>
       <EditorVotingType v-model="proposal" :voting-types="votingTypes as VoteType[]" />
-      <EditorChoices v-model="proposal" :definition="CHOICES_DEFINITION" />
+      <EditorChoices
+        v-model="proposal"
+        :voting-types="votingTypes as VoteType[]"
```

Comment:
We shouldn't map it `as VoteType[]` - it can also be null.

I'd centralise where we decide if we should show this part of UI here instead of inside components (I don't think it makes sense to have component that decides itself it it should render itself or not), like this:
```diff
diff --git a/apps/ui/src/components/EditorChoices.vue b/apps/ui/src/components/EditorChoices.vue
index aebd8697..c11eeaf6 100644
--- a/apps/ui/src/components/EditorChoices.vue
+++ b/apps/ui/src/components/EditorChoices.vue
@@ -1,17 +1,16 @@
 <script setup lang="ts">
 import Draggable from 'vuedraggable';
-import { Draft, VoteType } from '@/types';
+import { Draft } from '@/types';
 
 const proposal = defineModel<Draft>({ required: true });
 
 defineProps<{
-  votingTypes: VoteType[];
   definition: any;
 }>();
 </script>
 
 <template>
-  <div v-if="votingTypes.length > 1 || votingTypes[0] !== 'basic'" class="s-base mb-5">
+  <div class="s-base mb-5">
     <h4 class="eyebrow mb-2.5">Choices</h4>
     <div class="flex flex-col gap-[10px]">
       <Draggable
diff --git a/apps/ui/src/components/EditorVotingType.vue b/apps/ui/src/components/EditorVotingType.vue
index 60f3c319..4dfc8fea 100644
--- a/apps/ui/src/components/EditorVotingType.vue
+++ b/apps/ui/src/components/EditorVotingType.vue
@@ -45,29 +45,27 @@ function handleVoteTypeSelected(type: VoteType) {
 </script>
 
 <template>
-  <template v-if="votingTypes.length > 1 || votingTypes[0] !== 'basic'">
-    <div class="s-base mb-5">
-      <h4 class="eyebrow mb-2.5">Voting type</h4>
-      <div class="flex flex-col gap-[12px]">
-        <UiSelector
-          v-for="(type, index) in votingTypes"
-          :key="index"
-          :is-active="proposal.type === type"
-          @click="handleVoteTypeSelected(type as VoteType)"
-        >
-          <img
-            :src="VOTING_TYPES_INFO[type].image"
-            :alt="VOTING_TYPES_INFO[type].label"
-            class="w-[122px] hidden sm:block shrink-0"
-          />
-          <div class="grow">
-            <span class="text-skin-heading">{{ VOTING_TYPES_INFO[type].label }}</span>
-            <div>
-              {{ VOTING_TYPES_INFO[type].description }}
-            </div>
+  <div class="s-base mb-5">
+    <h4 class="eyebrow mb-2.5">Voting type</h4>
+    <div class="flex flex-col gap-[12px]">
+      <UiSelector
+        v-for="(type, index) in votingTypes"
+        :key="index"
+        :is-active="proposal.type === type"
+        @click="handleVoteTypeSelected(type as VoteType)"
+      >
+        <img
+          :src="VOTING_TYPES_INFO[type].image"
+          :alt="VOTING_TYPES_INFO[type].label"
+          class="w-[122px] hidden sm:block shrink-0"
+        />
+        <div class="grow">
+          <span class="text-skin-heading">{{ VOTING_TYPES_INFO[type].label }}</span>
+          <div>
+            {{ VOTING_TYPES_INFO[type].description }}
           </div>
-        </UiSelector>
-      </div>
+        </div>
+      </UiSelector>
     </div>
-  </template>
+  </div>
 </template>
diff --git a/apps/ui/src/helpers/constants.ts b/apps/ui/src/helpers/constants.ts
index a8cbe293..6af8c1e1 100644
--- a/apps/ui/src/helpers/constants.ts
+++ b/apps/ui/src/helpers/constants.ts
@@ -23,4 +23,4 @@ export const COINGECKO_BASE_ASSETS = {
 
 export const MAX_SYMBOL_LENGTH = 12;
 export const BASIC_CHOICES = ['For', 'Against', 'Abstain'];
-export const SUPPORTED_VOTING_TYPES = ['basic', 'single-choice', 'approval'];
+export const SUPPORTED_VOTING_TYPES = ['basic', 'single-choice', 'approval'] as const;
diff --git a/apps/ui/src/views/Editor.vue b/apps/ui/src/views/Editor.vue
index 46e02426..d0b20de4 100644
--- a/apps/ui/src/views/Editor.vue
+++ b/apps/ui/src/views/Editor.vue
@@ -340,12 +340,10 @@ export default defineComponent({
         />
         <UiLinkPreview :key="proposalKey || ''" :url="proposal.discussion" />
       </div>
-      <EditorVotingType v-model="proposal" :voting-types="votingTypes as VoteType[]" />
-      <EditorChoices
-        v-model="proposal"
-        :voting-types="votingTypes as VoteType[]"
-        :definition="CHOICES_DEFINITION"
-      />
+      <template v-if="votingTypes && (votingTypes.length > 1 || votingTypes[0] !== 'basic')">
+        <EditorVotingType v-model="proposal" :voting-types="votingTypes" />
+        <EditorChoices v-model="proposal" :definition="CHOICES_DEFINITION" />
+      </template>
       <div
         v-if="
           space &&
```